modules/smtp/modules.php

```diff
@@ -1896,8 +1896,8 @@ function save_imap_draft($atts, $id, $session, $mod, $mod_cache, $uploaded_files
         // Convert all header keys to lowercase
         $msg_header_lower = array_change_key_case($msg_header, CASE_LOWER);
         $mime_headers_lower = array_change_key_case($mime->get_headers(), CASE_LOWER);
-        if (isset($msg_header_lower['message-id'], $mime_headers_lower['message-id'])) {
-            if ($msg_header_lower['message-id'] === $mime_headers_lower['message-id']) {
+        if (isset($msg_header_lower['message-id'], $mime_headers_lower['message-id']) || isset($msg_header_lower['message_id'], $mime_headers_lower['message_id'])) {
+            if (($msg_header_lower['message-id'] === $mime_headers_lower['message-id']) || ($msg_header_lower['message_id'] === $mime_headers_lower['message_id'])) {
```

Comment:
Better divide into 2 if statements for both message_id and message-id cases - if isset && === for the first one and then for the second one. Since you return in case of true value, it is OK to do the ifs sequentially one after the other. The current code risks of using message-id when it is not set or message_id when it is not set.

Reply:
copy that @kroky I made some changes there.